### ISSUE TITLE
Woocommerce Accessibility Issues And Problems

### ISSUE THREAD

**Issue by javad2625** [woocommerce/woocommerce#13288]
Hello
I am a blind Woocommerce user who uses JAWS screen reader for reading everything on the computer screen. I like Woocommerce very much and I use it for my online shop because, just like Wordpress, Woocommerce respects accessibility standards very strictly, so blind users can use it too, but there are a few issues in the places that we are forced to use drag&drop or mouse, and we cannot do the task with keyboard:
1- In WC config, in payment methods, people can drag and drop the various payment methods in order to change their priorities, but blind users cannot, because they are not able to use their mouse. We should have an alternative method for adjusting the payment priorities, so blind users can also click on a few links, checkboxes or standard text-based elements for changing the priorities. Even if we cannot change the priorities, it will be ideal if we have an option for determining the default payment option. For example, if I will be able to adjust "Mellat Bank" as the default payment method, this will solve my problem. Right now, I have to disable all other payment methods except Bank Mellat, in order to make it the default option.

2- If I adjust a product type to "Variable Product", I will not be able to adjust the price of each variable without using my mouse. For example, I am able to define 3 variables, but I am not able to adjust the price of each variable without using my mouse. The properties of each variable will appear only if we click by its side with the mouse pointer, but not with our keyboard.

Thank you
www.irannopendar.com

**Comment by mikejolley** [comment]
We can take care of 2 during #13084. The whole UI there needs looking at.

For 1, would it help if we allowed the sorting icon to be clicked/selected via keyboard with a dialog to choose position?

**Comment by javad2625** [comment]
> For 1, would it help if we allowed the sorting icon to be clicked/selected via keyboard with a dialog to choose position?

Let me see if I understand your sentence rightly. You mean for example, there will be a "sort" link beside each payment gateway. For instance, there is a "sort" link or "sort" button" beside "Bank Mellat". I click "sort" link, and a dialogue box asks me the priority of this gateway. For example, I select it as "Priority 1", just as I select the priority of widgets in a sidebar. If this is what you mean, yes it will definitely solve our problem.  

Let me tell you about the widgets page. There is exactly the same problem in the widgets area of Wordpress. We have to use drag&drop, or we will not be able to add any widgets to any of the sidebars. Blind users cannot do drag&drop.  Wordpress has solved this problem with the utmost care about accessibility standards. When we go to "Screen Options" of widgets area, it has an icon named "Enable Accessibility Mode". If we enable "Accessible Mode", a link appears beside each widget. For example "Add" or "Edit" links! We can easily click/enter the link and open the widget. We can also select the location and priority of our widget via radio-buttons and comboboxes. That is very ideal and easy to use. Blind people are very convenient with standard text-based HTML elements such as links, buttons, comboboxes, radio-bottons and checkboxes, but they are not convenient with clickable items such as clickable graphics or drag&drop listboxes, unless those clickable graphics or listboxes follow accessibility standards as well.

**Comment by javad2625** [comment]
> For 1, would it help if we allowed the sorting icon to be clicked/selected via keyboard with a dialog to choose position?

Let me see if I understand your sentence rightly. You mean for example, there will be a "sort" link beside each payment gateway. For instance, there is a "sort" link or "sort" button" beside "Bank Mellat". I click "sort" link, and a dialogue box asks me the priority of this gateway. For example, I select it as "Priority 1", just as I select the priority of widgets in a sidebar. If this is what you mean, yes it will definitely solve our problem.  

Let me tell you about the widgets page. There is exactly the same problem in the widgets area of Wordpress. We have to use drag&drop, or we will not be able to add any widgets to any of the sidebars. Blind users cannot do drag&drop.  Wordpress has solved this problem with the utmost care about accessibility standards. When we go to "Screen Options" of widgets area, it has an icon named "Enable Accessibility Mode". If we enable "Accessible Mode", a link appears beside each widget. For example "Add" or "Edit" links! We can easily click/enter the link and open the widget. We can also select the location and priority of our widget via radio-buttons and comboboxes. That is very ideal and easy to use. Blind people are very convenient with standard text-based HTML elements such as links, buttons, comboboxes, radio-bottons and checkboxes, but they are not convenient with clickable items such as clickable graphics or drag&drop listboxes, unless those clickable graphics or listboxes follow accessibility standards as well.

**Comment by javad2625** [comment]
> For 1, would it help if we allowed the sorting icon to be clicked/selected via keyboard with a dialog to choose position?

Let me see if I understand your sentence rightly. You mean for example, there will be a "sort" link beside each payment gateway. For instance, there is a "sort" link or "sort" button" beside "Bank Mellat". I click "sort" link, and a dialogue box asks me the priority of this gateway. For example, I select it as "Priority 1", just as I select the priority of widgets in a sidebar. If this is what you mean, yes it will definitely solve our problem.  

Let me tell you about the widgets page. There is exactly the same problem in the widgets area of Wordpress. We have to use drag&drop, or we will not be able to add any widgets to any of the sidebars. Blind users cannot do drag&drop.  Wordpress has solved this problem with the utmost care about accessibility standards. When we go to "Screen Options" of widgets area, it has an icon named "Enable Accessibility Mode". If we enable "Accessible Mode", a link appears beside each widget. For example "Add" or "Edit" links! We can easily click/enter the link and open the widget. We can also select the location and priority of our widget via radio-buttons and comboboxes. That is very ideal and easy to use. Blind people are very convenient with standard text-based HTML elements such as links, buttons, comboboxes, radio-bottons and checkboxes, but they are not convenient with clickable items such as clickable graphics or drag&drop listboxes, unless those clickable graphics or listboxes follow accessibility standards as well.

**Comment by WPprodigy** [comment]
Ohh, accessibility mode for widgets is really nice. Never knew that was there!

Having a sort link that triggers a selection for the Position/Priority would seem to follow the WP way then.

Another option though could be to have 2 extra buttons, an up and a down arrow beside each gateway. Clicking them would +1 or -1 the position.

**Comment by mikejolley** [comment]
Since variations is being looked at elsewhere, I'm going to move this to a new issue specifically for payment method drag and drop.
